rotkehlchen/chain/evm/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,183 @@
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _maybe_update_events_legacy_contrats(
+        decoded_events: list['EvmEvent'],
+        sender: ChecksumEvmAddress,
+        source_asset: CryptoAsset,
+        destination_asset: CryptoAsset,
+        spent_amount: FVal,
+        return_amount: FVal,
+        notify_user: Callable[['EvmEvent', str], None],
+) -> None:
+    """
+    Use the information from a trade transaction to modify the HistoryEvents from receive/send to
+    trade if the conditions are correct.
+    """
+    in_event = out_event = None
+    for event in decoded_events:
+        try:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+        except (UnknownAsset, WrongAssetType):
+            notify_user(event, CPT_KYBER)
+            continue
+        # it can happen that a spend event get decoded first by an amm decoder. To make sure that
+        # the event matches we check both event type and subtype
+        if (event.event_type == HistoryEventType.SPEND or event.event_subtype == HistoryEventSubType.SPEND) and event.location_label == sender and event.asset == source_asset and event.balance.amount == spent_amount:  # noqa: E501
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.SPEND
+            event.counterparty = CPT_KYBER
+            event.notes = f'Swap {event.balance.amount} {crypto_asset.symbol} in kyber'
+            out_event = event
+        elif (event.event_type == HistoryEventType.RECEIVE or event.event_subtype == HistoryEventSubType.RECEIVE) and event.location_label == sender and event.balance.amount == return_amount and destination_asset == event.asset:  # noqa: E501
+            event.event_type = HistoryEventType.TRADE
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.counterparty = CPT_KYBER
+            event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} from kyber swap'
+            in_event = event
+
+        maybe_reshuffle_events(ordered_events=[out_event, in_event], events_list=decoded_events)
+
+
+class KyberCommonDecoder(DecoderInterface):
+
+    def _legacy_contracts_basic_info(self, tx_log: EvmTxReceiptLog) -> tuple[ChecksumEvmAddress, CryptoAsset, CryptoAsset]:  # noqa: E501
+        """
+        Returns:
+        - address of the sender
+        - source token (can be none)
+        - destination token (can be none)
+        May raise:
+        - DeserializationError when using hex_or_bytes_to_address
+        """
+        sender = hex_or_bytes_to_address(tx_log.topics[1])
+        source_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+        destination_token_address = hex_or_bytes_to_address(tx_log.data[32:64])
+
+        source_token = self.base.get_or_create_evm_asset(source_token_address)
+        destination_token = self.base.get_or_create_evm_asset(destination_token_address)
+        return sender, source_token, destination_token
```

Comment:
as mentioned in discord legacy is only for ethereum. `KyberDecoder` on ethereum should have this logic since is not used in the other evm chains